File: app/src/main/java/com/anbillon/lunarlite/di/module/AppModule.java
package com.anbillon.lunarlite.di.module;

import android.content.Context;
import com.anbillon.lunarlite.app.LunarApplication;
import dagger.Module;
import dagger.Provides;
import javax.inject.Singleton;

/**
 * Dagger module that provides app level related collaborators.
 *
 * @author Vincent Cheung (coolingfall@gmail.com)
 */
@Module(includes = DbModule.class) public final class AppModule {
	private final LunarApplication lunarApplication;

	public AppModule(LunarApplication lunarApplication) {
		this.lunarApplication = lunarApplication;
	}

	@Provides @Singleton Context provideAppContext() {
		return lunarApplication.getApplicationContext();
	}
}

File: app/src/main/java/com/anbillon/lunarlite/di/module/AppModule.kt
package com.anbillon.lunarlite.di.module

import android.content.Context
import com.anbillon.lunarlite.app.LunarApplication
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Application module used in this project.
 *
 * @author Vincent Cheung (coolingfall@gmail.com)
 */
@Module(includes = arrayOf(DbModule::class))
class AppModule(
    private val application: LunarApplication) {

  @Provides
  @Singleton internal fun provideAppContext(): Context {
    return application.applicationContext
  }
}
